lib/hdb/hdb.h

```diff
@@ -80,9 +80,18 @@ enum hdb_lockop{ HDB_RLOCK, HDB_WLOCK };
 #define HDB_CAP_F_SHARED_DIRECTORY      8
 
 /* auth status values */
-#define HDB_AUTH_SUCCESS		0
+#define HDB_AUTHZ_SUCCESS		0
```

Comment:
Is it `AUTH` or `AUTHZ`?

Also, `#define HDB_AUTH_UNKNOWN -1`?

Reply:
AUTHZ is because it was for authorization, eg it all worked, not just that it was authenticated with a password (for the password check hooks, which need to reset the bad pwd count).

Reply:
OK, then shouldn't the others be prefixed with `HDB_AUTHZ_` too? Looks a bit inconsistent. And perhaps the callback should be `hdb_authz_status()`?

Reply:
No, the rest are about authentication, not authorization.

Reply:
Well except the GSS bit, but that was just cargo-culted from PKINIT as none of us have really taken the time to pick out how that fits into this.

Reply:
GSS is also an authentication mechanism, in this context.

Reply:
I get what you're saying but `AUTHZ` just one case looks like a typo to me, but I don't mind too much. What do you think @nicowilliams?

Reply:
I can add comments to `hdb.h` if that helps?

Reply:
deferring to @nicowilliams

Reply:
eg 
```
diff --git a/lib/hdb/hdb.h b/lib/hdb/hdb.h
index 938b207034..4252e119b8 100644
--- a/lib/hdb/hdb.h
+++ b/lib/hdb/hdb.h
@@ -80,17 +80,17 @@ enum hdb_lockop{ HDB_RLOCK, HDB_WLOCK };
 #define HDB_CAP_F_SHARED_DIRECTORY      8
 
 /* auth status values */
-#define HDB_AUTHZ_SUCCESS              0
+#define HDB_AUTHZ_SUCCESS              0 /* Overall authorization success */
 #define HDB_AUTH_WRONG_PASSWORD                1
 #define HDB_AUTH_INVALID_SIGNATURE     2
-#define HDB_AUTH_CORRECT_PASSWORD      3
+#define HDB_AUTH_CORRECT_PASSWORD      3 /* Authentication success with a password */
 #define HDB_AUTH_CLIENT_UNKNOWN        4
 #define HDB_AUTH_CLIENT_LOCKED_OUT     5
 #define HDB_AUTH_GENERIC_SUCCESS       6
 #define HDB_AUTH_GENERIC_FAILURE       7
-#define HDB_AUTH_PKINIT_SUCCESS        8
+#define HDB_AUTH_PKINIT_SUCCESS        8 /* PKINIT authentication success */
 #define HDB_AUTH_PKINIT_FAILURE        9
-#define HDB_AUTH_GSS_SUCCESS           10
+#define HDB_AUTH_GSS_SUCCESS           10  /* GSS authentication success */
 #define HDB_AUTH_GSS_FAILURE           11
 
 /* key usage for master key */
```

Reply:
Could have HDB_AUTHZ_UNKNOWN -1 please?

Reply:
Also, again, really this doesn't make sense to me. It's conflating authorisation and authentication, they're in my mind orthogonal not subsets. I'd prefer more verbiage if it's clear what it means. e.g. `HDB_AUTHZ_AUTH_FAILED~WRONG_PASSWORD`.

Reply:
I've added comments.

Reply:
My take on the HDB_AUTH vs HDB_AUTHZ confusion is that these are constants that are supposed to describe auth_status values.  They are not intended to be general HDB constants.  Therefore, I believe the prefix should be tied to the name of the field. 

Perhaps HDB_AUTHSTATUS or something similar.  If it is so important to distinguish between AUTHN and AUTHZ then perhaps the field name needs to be changed as well.

Reply:
> Could have HDB_AUTHZ_UNKNOWN -1 please?

I would prefer UNKNOWN to be zero.

Reply:
That sounds good, for the other constants too?

Reply:
Yes, they are consistent.